start_signon.sh

```diff
@@ -30,5 +30,5 @@ bundle install --path ${APP_ROOT}_bundle
 
 ${GEM_ROOT}/stop_signon.sh
 
-RAILS_ENV=test bundle exec rake db:drop db:create db:schema:load
+DISABLE_DATABASE_ENVIRONMENT_CHECK=1 RAILS_ENV=test bundle exec rake db:reset
```

Comment:
I was having some trouble with the new Rails 5 stuff failing my builds  on  https://github.com/alphagov/gds-sso/pull/123

I think it was because on the CI boxes, the database were not properly upgraded and were either missing or had the wrong version of the `ar_internal_metadata` table. I found that doing this https://github.com/alphagov/gds-sso/blob/test-update-signon-to-bin-rails/start_signon.sh#L35 coerces the `value` field for the environment to `test`

However, once the database actually has the `ar_internal_metadata` and the correct environment, I'm not sure we need to `DISABLE_DATABASE_ENVIRONMENT_CHECK` anymore?  Keen to pick your brains on this anyway :)

Reply:
@bevanloon sure, unfortunately the link there doesn't work so I'm not sure actually what happened in your scenario above.

I've had to deal with this on a few apps of ours and found setting the env var is the only way to reliably be able to deal with the db not existing - and also means you don't need to do any env set stuff (that will fail if db doesn't exist)

Reply:
oh yah - the branch was deleted :man_facepalming: 

I did this, before before the whole `db:...` dance:

```
bin/rails db:environment:set RAILS_ENV=test
```
which seemed to work nicely, but I see in puppet(https://github.com/alphagov/govuk-puppet/commit/125038f584bfceac66ad069a457893e2719330fe), you've already bumped into this. it's all somewhat annoying, but anyway.....

Reply:
@bevanloon yeah it's a bit of an unfortunate level of pain really for something so simple.